crawl-ref/source/timed-effects.cc

```diff
@@ -200,6 +200,18 @@ static void _abyss_speed(int /*time_delta*/)
         --you.abyss_speed;
 }
 
+//Drain int until at 1 base int; disregarding gear.
+//Rate of drain is proportional to potential remaining drain
+static void _ghoul_hunger(int /*time_delta*/)
```

Comment:
This should not be implemented as a timed-effect, but in `player_reacts`, since it occurs only when monsters are in LOS and is a "player reaction" like regeneration etc. It will behave strangely if implemented as a timed effect.

Additionally it should scale the chance based on `time_taken` so that the chance doesn't vary with fast/slow actions.